flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/InternalTimeServiceManager.java

```diff
@@ -51,6 +52,21 @@ <N> InternalTimerService<N> getInternalTimerService(
             TypeSerializer<N> namespaceSerializer,
             Triggerable<K, N> triggerable);
 
+    /**
+     * Creates an {@link InternalTimerServiceAsyncImpl} for handling a group of timers identified by
+     * the given {@code name}. The timers are scoped to a key and namespace. Mainly used by async
+     * operators.
+     *
+     * <p>Some essential order preservation will be added when the given {@link Triggerable} is
+     * invoked.
+     */
+    <N> InternalTimerService<N> getAsyncInternalTimerService(
+            String name,
+            TypeSerializer<K> keySerializer,
+            TypeSerializer<N> namespaceSerializer,
+            Triggerable<K, N> triggerable,
+            AsyncExecutionController asyncExecutionController);
```

Comment:
It is better to provide the type parameter `K` from here and any other methods of this class.